### PR TITLE
WORKFLOW_METADATA_SUMMARY_ENTRY updates for new archiving mode [BW-561]

### PR DESCRIPTION
**Pull request by cjllanwarne**
Note: We **must** wait for https://broadworkbench.atlassian.net/browse/BW-548 to be deployed before we merge this one.

### CODE REVIEW

**Review by aednichols — COMMENTED** [review]
Just a pregull idea you could work on while you wait on this PR's dependency:

It would be good to have an estimate for how long this migration takes on prod, to provide to the folks who set maintenance windows and set releases.

This _should be_ very fast but it wouldn't be bad to test on a prod clone. (When I created a similar index, it took about a minute.)

File: database/migration/src/main/resources/metadata_changesets/reset_archive_statuses_to_null.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+    <changeSet id="reset_archive_statuses_to_null" author="cjllanwarne" dbms="hsqldb,mariadb,mysql,postgresql">
+        <update tableName="WORKFLOW_METADATA_SUMMARY_ENTRY">
+            <!-- Note: Omiting the value here updates it to NULL -->
+            <column name="METADATA_ARCHIVE_STATUS"/>
+        </update>
+    </changeSet>
+    
+</databaseChangeLog>
```
Comment (cjllanwarne) [inline database/migration/src/main/resources/metadata_changesets/reset_archive_statuses_to_null.xml:12]: Add newline

**Comment by cjllanwarne** [comment]
Note to self: Make sure a 1-2 minute startup while the migration happens will not cause k8s liveness check issues

**Review by panentheos — APPROVED** [review]
Looks good.

**Review by aednichols — APPROVED** [review]

**Pull request merged**